<launch>
   <arg name="particle_num"                                    default="1000"/>

   <group ns="PF_nav">
      <node pkg="controller_tutorial" name="controller_tutorial" type="controller_tutorial" output="screen">

         <!-- 制御周期 Hz -->
         <param name="control_frequency"                       value="50.0"/>

         <!-- ノイズパラメータ -->
         <param name="norm_noise_mean_x"                       value="0.0"/>
         <param name="norm_noise_variance_x"                   value="0.0001"/>
         <param name="norm_noise_mean_y"                       value="0.0"/>
         <param name="norm_noise_variance_y"                   value="0.0001"/>
         <param name="norm_noise_mean_yaw"                     value="0.0"/>
         <param name="norm_noise_variance_yaw"                 value="0.0001"/>
         <param name="norm_noise_mean_linear_velocity"         value="0.0"/>
         <param name="norm_noise_variance_linear_velocity"     value="0.0001"/>
         <param name="norm_noise_mean_angular_velocity"        value="0.0"/>
         <param name="norm_noise_variance_angular_velocity"    value="0.0001"/>
         
         <!-- 移動制御のパラメータ -->
         <param name="gain_p"                                  value="1.0"/>
         <param name="gain_i"                                  value="0.1"/>
         <param name="gain_d"                                  value="0.001"/>
         <param name="stop_margin_angle"                       value="0.01"/>
         <param name="stop_margin_distance"                    value="0.03"/>
         <param name="max_linear_velocity"                     value="0.5"/>
         <param name="max_angular_velocity"                    value="0.6"/>
      </node>

      <node pkg="PF_nav" name="PF_nav_MarkerArray" type="PF_nav_MarkerArray" output="screen">
         <rosparam command="load" file="$(find PF_nav)/yaml/markers.yaml"/>
      </node>
     
     <node pkg="PF_nav" name="PF_nav_Visualization" type="PF_nav_Visualization" output="screen">
         <!-- <param name="particle_num"                            value="$(arg particle_num)"/> -->

         <!-- 生成するパーティクル数 -->
         <param name="particle_num"                            value="$(arg particle_num)"/>

         <!-- パーティクル移動時のノイズパラメータ -->
         <param name="norm_noise_mean_linear_velocity"         value="0.0"/>
         <param name="norm_noise_variance_linear_velocity"     value="0.1"/>
         <param name="norm_noise_mean_angular_velocity"        value="0.0"/>
         <param name="norm_noise_variance_angular_velocity"    value="0.1"/>
         
         <!-- ロボット観測値のノイズパラメータ -->
         <param name="norm_noise_mean_Scan_distance"         value="0.0"/>
         <param name="norm_noise_variance_Scan_distance"     value="0.000856"/>
         <param name="norm_noise_mean_Scan_angle"        value="0.0"/>
         <param name="norm_noise_variance_Scan_angle"    value="0.00464"/>
     </node>

      <node pkg="rviz" type="rviz" name="rviz" required="true" args="-d $(find PF_nav)/rviz/PF_nav.rviz"/>

   </group>
</launch>
